src/Stratis.Bitcoin/P2P/PeerConnectorDiscovery.cs

```diff
@@ -40,7 +40,7 @@ public sealed class PeerConnectorDiscovery : PeerConnector
         /// <inheritdoc/>
         public override void OnInitialize()
         {
-            this.MaximumNodeConnections = 8;
+            this.MaxOutboundConnections = this.NodeSettings.ConfigReader.GetOrDefault("maxOutboundConnections", 8);
```

Comment:
Should we maybe not set this to a property on NodeSettings? I'm just worried that if we don't, this property will get "buried" i.e. a future dev or user won't know that this configurable element of the node exists, unless they go dig in peer connector code? What do you think?

Reply:
> I'm just worried that if we don't, this property will get "buried" i.e. a future dev or user won't know that this configurable element of the node exists

As far as I know Jeremy wanted to completely change the approach we take to the config file (there is a card on trello for this) so after he does that we will have a json config file initialized with default values so it will be pretty straight forward to understand what can be configured and what can't.
This should solve the problem of  all "buried" properties, not just this one

Reply:
OK all good then @noescape00. Can we maybe then just add a TODO comment so that we remember for the future?

Reply:
Sure, fixed.